source/prtcl/prtcl_main.c

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright (C) 2023 by Ivy Bowling <motolegacy.git@gmail.com>
+//
+// Permission to use, copy, modify, and/or distribute this software 
+// for any purpose with or without fee is hereby granted.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL 
+// WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED 
+// WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL 
+// THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR 
+// CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM 
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, 
+// NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN 
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+//
+
+#include <stdio.h>
+#include <string.h>
+
+#include <util.h>
+#include <prtcl/protocol_file.h>
+
+char* PRTCL_RetrieveDocument(char* req_uri)
+{
+    // Check the protocol identifier so we know
+    // how to handle retrieving the document.
+    // file://
+    if (req_uri[0] == 'f' && req_uri[1] == 'i' &&
```

Comment:
This works yes, but will get messy the more protocols we support I feel.

Reply:
Can you suggest an alternative? Protocols have inconsistent lengths so we cannot just build a chunk at whatever coordinate. So the only other solution I can fathom would be to walk through the string first to find "://" to know the index at which to end the split, which sounds a lot less elegant than what is being done here.